### ISSUE TITLE
Error running lstchain_data_r0_to_dl1

### ISSUE THREAD

**Issue by rlopezcoto** [cta-observatory/cta-lstchain#386]
@FrancaCassol was doing some tests after the last merge of master and found a problem. I could not reproduced it using the latest data:
`lstchain_data_r0_to_dl1 -f ~/Downloads/LST-1.1.Run02034.0000.fits.fz --pedestal-file ~/Downloads/drs4_pedestal.Run2005.0000.fits --calibration-file ~/Downloads/calibration.Run2006.0000.hdf5 --time-calibration ~/Downloads/time_calibration.Run2006.0000.hdf5`

but it was there when reading older data:
```
lstchain_data_r0_to_dl1 -f ~/Downloads/LST-1.4.Run01566.0326.fits.fz --pedestal-file ~/Downloads/drs4_pedestal.Run2005.0000.fits --calibration-file ~/Downloads/calibration.Run2006.0000.hdf5 --time-calibration ~/Downloads/time_calibration.Run2006.0000.hdf5 
/Users/rubenlopez/anaconda/envs/cta-dev/lib/python3.6/site-packages/corsikaio/subblocks/dtypes.py:20: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  return np.dtype(dict(**dt))
EXTRACTOR <ctapipe.image.extractor.LocalPeakWindowSum object at 0x1a2ac22470>
EXTRACTOR <ctapipe.image.extractor.GlobalPeakWindowSum object at 0x1088385f8>
USING FILTERS:  Filters(complevel=5, complib='blosc:zstd', shuffle=True, bitshuffle=False, fletcher32=True, least_significant_digit=None)
0
Traceback (most recent call last):
  File "/Users/rubenlopez/anaconda/envs/cta-dev/lib/python3.6/site-packages/astropy/units/quantity_helper/converters.py", line 181, in converters_and_unit
    if can_have_arbitrary_unit(args[i]):
  File "/Users/rubenlopez/anaconda/envs/cta-dev/lib/python3.6/site-packages/astropy/units/quantity_helper/converters.py", line 122, in can_have_arbitrary_unit
    return np.all(np.logical_or(np.equal(value, 0.), ~np.isfinite(value)))
TypeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/rubenlopez/anaconda/envs/cta-dev/bin/lstchain_data_r0_to_dl1", line 11, in <module>
    load_entry_point('lstchain', 'console_scripts', 'lstchain_data_r0_to_dl1')()
  File "/Users/rlopezcoto/Code/ctasoft/fork/cta-lstchain/lstchain/scripts/lstchain_data_r0_to_dl1.py", line 151, in main
    tib_counter0=args.tib_counter0
  File "/Users/rlopezcoto/Code/ctasoft/fork/cta-lstchain/lstchain/reco/r0_to_dl1.py", line 467, in r0_to_dl1
    ucts_time_utc = unix_tai_to_utc(ucts_time)
  File "/Users/rlopezcoto/Code/ctasoft/fork/cta-lstchain/lstchain/reco/utils.py", line 480, in unix_tai_to_utc
    return UCTS_EPOCH + u.Quantity(timestamp, u.s, copy=False)
  File "/Users/rubenlopez/anaconda/envs/cta-dev/lib/python3.6/site-packages/astropy/units/quantity.py", line 444, in __array_ufunc__
    converters, unit = converters_and_unit(function, method, *inputs)
  File "/Users/rubenlopez/anaconda/envs/cta-dev/lib/python3.6/site-packages/astropy/units/quantity_helper/converters.py", line 196, in converters_and_unit
    for arg in args])))
TypeError: Unsupported operand type(s) for ufunc add: 'Time,Quantity'
```

@morcuended can you reproduce it?

**Comment by morcuended** [comment]
Checking

**Comment by morcuended** [comment]
> @FrancaCassol was doing some tests after the last merge of master and found a problem. I could not reproduced it using the latest data:
> lstchain_data_r0_to_dl1 -f ~/Downloads/LST-1.1.Run02034.0000.fits.fz --pedestal-file ~/Downloads/drs4_pedestal.Run2005.0000.fits --calibration-file ~/Downloads/calibration.Run2006.0000.hdf5 --time-calibration ~/Downloads/time_calibration.Run2006.0000.hdf5

Actually by doing that I got:

```
  File "/fefs/home/daniel.morcuende/software/cta-lstchain/cta-lstchain/lstchain/reco/r0_to_dl1.py", line 610, in r0_to_dl1
    table.write(muon_output_filename, format='fits', overwrite=True)
  File "/home/daniel.morcuende/.local/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/table/connect.py", line 114, in __call__
    registry.write(instance, *args, **kwargs)
  File "/home/daniel.morcuende/.local/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/io/registry.py", line 566, in write
    writer(data, *args, **kwargs)
  File "/home/daniel.morcuende/.local/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/io/fits/connect.py", line 396, in write_table_fits
    table_hdu.writeto(output)
  File "/home/daniel.morcuende/.local/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/utils/decorators.py", line 521, in wrapper
    return function(*args, **kwargs)
  File "/home/daniel.morcuende/.local/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/io/fits/hdu/base.py", line 1583, in writeto
    checksum=checksum)
  File "/home/daniel.morcuende/.local/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/utils/decorators.py", line 521, in wrapper
    return function(*args, **kwargs)
  File "/home/daniel.morcuende/.local/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/io/fits/hdu/hdulist.py", line 919, in writeto
    fileobj = _File(fileobj, mode=mode, overwrite=overwrite)
  File "/home/daniel.morcuende/.local/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/utils/decorators.py", line 521, in wrapper
    return function(*args, **kwargs)
  File "/home/daniel.morcuende/.local/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/io/fits/file.py", line 178, in __init__
    self._open_filename(fileobj, mode, overwrite)
  File "/home/daniel.morcuende/.local/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/io/fits/file.py", line 555, in _open_filename
    self._file = fileobj_open(self.name, IO_FITS_MODES[mode])
  File "/home/daniel.morcuende/.local/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/io/fits/util.py", line 397, in fileobj_open
    return open(filename, mode, buffering=0)
FileNotFoundError: [Errno 2] No such file or directory: './muons_data/muons_LST-1.Run02034.0000.fits'
```

**Comment by morcuended** [comment]
I can reproduce the other error for older runs:

```bash
srun lstchain_data_r0_to_dl1 -f /fefs/aswg/data/real/R0/20191123/LST-1.1.Run01616.0000.fits.fz --pedestal-file /fefs/aswg/data/real/calibration/20191123/v00/drs4_pedestal.Run1611.0000.fits --calibration-file /fefs/aswg/data/real/calibration/20191123/v00/calibration.Run1614.0000.hdf5 --time-calibration /fefs/aswg/data/real/calibration/20191123/v00/time_calibration.Run1614.0000.hdf5
```
traceback error:

```python
Traceback (most recent call last):
  File "/home/daniel.morcuende/.local/miniconda3/envs/lst-dev/bin/lstchain_data_r0_to_dl1", line 11, in <module>
    load_entry_point('lstchain', 'console_scripts', 'lstchain_data_r0_to_dl1')()
  File "/fefs/home/daniel.morcuende/software/cta-lstchain/cta-lstchain/lstchain/scripts/lstchain_data_r0_to_dl1.py", line 151, in main
    tib_counter0=args.tib_counter0
  File "/fefs/home/daniel.morcuende/software/cta-lstchain/cta-lstchain/lstchain/reco/r0_to_dl1.py", line 467, in r0_to_dl1
    ucts_time_utc = unix_tai_to_utc(ucts_time)
  File "/fefs/home/daniel.morcuende/software/cta-lstchain/cta-lstchain/lstchain/reco/utils.py", line 480, in unix_tai_to_utc
    return UCTS_EPOCH + u.Quantity(timestamp, u.s, copy=False)
  File "/home/daniel.morcuende/.local/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/units/quantity.py", line 444, in __array_ufunc__
    converters, unit = converters_and_unit(function, method, *inputs)
  File "/home/daniel.morcuende/.local/miniconda3/envs/lst-dev/lib/python3.7/site-packages/astropy/units/quantity_helper/converters.py", line 196, in converters_and_unit
    for arg in args])))
TypeError: Unsupported operand type(s) for ufunc add: 'Time,Quantity'
```

**Comment by MaxNoe** [comment]
You sure this is a problem with the run itself and not the astropy / numpy version you are using?

Time should be addable to a quantity with unit seconds.

**Comment by morcuended** [comment]
```
>>> numpy.__version__
'1.18.1'
>>> astropy.__version__
'3.2.3'
```

**Comment by MaxNoe** [comment]
So a simle example works fine:

```
In [1]: import astropy.units as u                                                                                                             

In [2]: from astropy.time import Time                                                                                                         

In [3]: Time.now() + 5 * u.s                                                                                                                  
Out[3]: <Time object: scale='utc' format='datetime' value=2020-04-30 16:32:34.595050>
```

Can you print the two values? `UCTS_EPOCH` and the `timestamp`?

**Comment by MaxNoe** [comment]
Ah wait:

See this:
```
TypeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```

I think you have a `nan` there in the timestamp.

**Comment by MaxNoe** [comment]
Jep:
```
In [5]: Time.now() + np.nan * u.s                                                                                                             
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/.local/anaconda3/envs/lst/lib/python3.7/site-packages/astropy/units/quantity_helper/converters.py in converters_and_unit(function, method, *args)
    180                         continue
--> 181                     if can_have_arbitrary_unit(args[i]):
    182                         converters[i] = None

~/.local/anaconda3/envs/lst/lib/python3.7/site-packages/astropy/units/quantity_helper/converters.py in can_have_arbitrary_unit(value)
    121     """
--> 122     return np.all(np.logical_or(np.equal(value, 0.), ~np.isfinite(value)))
    123 

TypeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
<ipython-input-5-285622984270> in <module>
----> 1 Time.now() + np.nan * u.s

~/.local/anaconda3/envs/lst/lib/python3.7/site-packages/astropy/units/quantity.py in __array_ufunc__(self, function, method, *inputs, **kwargs)
    459         # consistent units between two inputs (e.g., in np.add) --
    460         # and the unit of the result (or tuple of units for nout > 1).
--> 461         converters, unit = converters_and_unit(function, method, *inputs)
    462 
    463         out = kwargs.get('out', None)

~/.local/anaconda3/envs/lst/lib/python3.7/site-packages/astropy/units/quantity_helper/converters.py in converters_and_unit(function, method, *args)
    194                                 "'{}'".format(function.__name__,
    195                                                ','.join([arg.__class__.__name__
--> 196                                                          for arg in args])))
    197 
    198         # In the case of np.power and np.float_power, the unit itself needs to

TypeError: Unsupported operand type(s) for ufunc add: 'Time,Quantity'
```

**Comment by morcuended** [comment]
Actually I was setting them to `nan` whenever the flag is not OK. I think in the previous version this was set to `-1 (* 1e-9)`

                            if event.lst.tel[telescope_id].evt.extdevices_presence & 2:
                                # UCTS presence flag is OK
                                ucts_time = event.lst.tel[telescope_id].evt.ucts_timestamp * 1e-9  # secs
                            else:
                                ucts_time = math.nan

That would be an easy solution, right?

**Comment by MaxNoe** [comment]
I'll prepare a PR

**Comment by MaxNoe** [comment]
Look what I found: https://github.com/astropy/astropy/issues/6509

Nearly three years old...

**Comment by morcuended** [comment]
In my last PR I changed the way `muon_output_filename` was created to:


```python
muon_output_filename = output_filename.replace("dl1", "muons").replace('LST-1.1', 'LST-1').replace('.h5', '')
```

the output_filename was in this case `./dl1_data/dl1_LST-1.Run02034.0000.fits` thus it changed also the name of the directory not only the basename.

```python
...
FileNotFoundError: [Errno 2] No such file or directory: './muons_data/muons_LST-1.Run02034.0000.fits'
```

So I'm not happy with the change I did. It may be prevented with something like:
```python
muon_output_filename = output_filename.replace("dl1_LST", "muons_LST").replace('LST-1.1', 'LST-1').replace('.h5', '')
```

**Comment by MaxNoe** [comment]
> Actually I was setting them to nan whenever the flag is not OK. I think in the previous version this was set to -1 (* 1e-9)

Let's just use a value of 0? That's easier to recognize and is invalid for all real LST data. Anyone will immediately recognize a value of 0 or 1970.

**Comment by rlopezcoto** [comment]
Solved in #388
